lib/src/platform/src/platform_object.dart

```diff
@@ -59,6 +58,29 @@ abstract class PlatformObject {
   }
 
   /// invoke platform method channel with AblyMessage encapsulation
+  /* FIXME(ikurek): This seems wrong - the structure is either
```

Comment:
I've done some digging about this, and it seems that there is an open issue https://github.com/ably/ably-flutter/issues/179 about this, but it's not a trivial change - as you mentioned, we would need to go trough all Objective-C and Java method handlers and refactor message handling in each of them. But I think it will be necessary sooner or later, since this structure seems to be a source of bugs - especially because `handle` value is often assigned to `message` field, leaving the handle as `null`

Reply:
@ikurek To avoid `TODO` / `FIXME` comments becoming a parallel, untracked backlog in the source code... please can you add to your comment a link to the issue you discovered.

Reply:
That's a good point, I've added a link to this issue in the codebase: https://github.com/ably/ably-flutter/pull/310/commits/039c78cae6540a0ef628afa35f009ad271b4c51f